### ISSUE TITLE
always_specify_types

### ISSUE THREAD

**Issue by zoechi** [dart-lang/linter#156]
from #144
> specialising generics like List

seems not to be implemented yet

```dart
final Map<EventType, async.StreamController> streamControllers =
      <EventType, async.StreamController>{};
```

does not complain about `StreamController> not being specialized (also not for `EventType` which is also generic in my case).

Dart VM version: 1.14.0-edge.a689bf41805c251b5bae646e0eb82f832d1e309f (Wed Nov 18 17:36:06 2015) on "linux_x64"

**Comment by pq** [comment]
Quite right!  Sorry for the slow reply.

https://codereview.chromium.org/1637813003/

**Comment by pq** [comment]
Implemented w/ bc9dcc27c7dddb0f6c36d7eadac9881e682395f2.

**Comment by zoechi** [comment]
nice :)
